lib/optimizely/custom_attribute_condition_evaluator.rb

```diff
@@ -133,20 +204,78 @@ def substring_evaluator(condition)
       condition_value = condition['value']
       user_provided_value = @user_attributes[condition['name']]
 
-      return nil unless user_provided_value.is_a?(String) && condition_value.is_a?(String)
+      unless user_provided_value.is_a?(String)
+        @logger.log(
+          Logger::WARN,
+          format(
+            Helpers::Constants::AUDIENCE_EVALUATION_LOGS['UNEXPECTED_TYPE'],
+            condition,
+            user_provided_value.class,
+            condition['name']
+          )
+        )
+        return nil
+      end
+
+      unless condition_value.is_a?(String)
+        @logger.log(
+          Logger::WARN,
+          format(Helpers::Constants::AUDIENCE_EVALUATION_LOGS['UNKNOWN_CONDITION_VALUE'], condition)
+        )
+        return nil
+      end
```

Comment:
Nit: would be slightly better (and certainly more consistent with other SDKs) to reorder the above two blocks. If the `condition_value` isn't a string, that'd be indicative of a broader problem that might be resolved in a future SDK, where it might actually be legitimate for the `user_provided_value` to be a non-string.

(I know we check for non-nil `user_provided_value` before we get to either of these blocks, but at least that helps us increase code re-use...)